core/src/duckdb/write.rs

```diff
@@ -707,6 +711,30 @@ fn write_to_table(
     Ok(rows as u64)
 }
 
+/// Executes an ANALYZE statement to update query optimizer statistics.
+/// This helps DuckDB's query planner generate better execution plans, especially after large data changes.
+/// https://duckdb.org/docs/stable/sql/statements/analyze
+///
+/// Errors are logged but do not fail the operation since statistics updates are non-critical.
+pub fn execute_analyze_sql(tx: &Transaction, table_name: &str) {
```

Comment:
`execute_analyze_sq` is public so it can be re-used, for example for own DuckDB write implementation like table-based partitioned write